### DISCUSSION TITLE
How to set TextInput to make it obvious in android app?

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1406] [Q&A]
When the background_color of the box is black, TextInput is not obvious in android app, how can it show obviously？

**Comment by mhsmith** [comment]
If you've set the `background_color`, then you'll probably want to set the `color` to something contrasting.

**Reply by ccinsz** [reply]
Yes, I try it, but it doesn't work. For example, make the main_box background_color="black", and make TextInput color="white". TextInput still show the black, cause not to see it. Only click it, it show the red underline.

**Reply by mhsmith** [reply]
OK, then you'll need to set the background color of the TextInput as well.

**Reply by ccinsz** [reply]
I try it, too. But it doesn't work. I wonder if it's a bug.

**Reply by freakboy3742** [reply]
To the best of our knowledge, setting color and background color works on TextInput, and we have a test suite validating this; however, we can't rule out there may be a bug. If you believe you've found a bug, please provide example code that reproduces the problem.

**Reply by ccinsz** [reply]
The example code is as follows:

import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW


class HelloWorld(toga.App):

    def startup(self):
        self.main_window = toga.MainWindow(title=self.app)
        label_style = Pack(font_size=16, padding_top=10, padding_right=10, color="white")

        label1 = toga.Label("Set label1：", style=label_style)
        self.textinput1 = toga.TextInput(placeholder="input the number",
                                         style=Pack(font_size=16, color="white"))

        label2 = toga.Label("Set label2：", style=label_style)
        self.textinput2 = toga.TextInput(placeholder="input the number",
                                         style=Pack(font_size=16, background_color="white"))

        label3 = toga.Label("Set label3：", style=label_style)
        self.textinput3 = toga.TextInput(placeholder="input the number",
                                         style=Pack(font_size=16, color="white", background_color="white"))

        main_box = toga.Box(children=[label1, self.textinput1, label2, self.textinput2, label3, self.textinput3],
                            style=Pack(flex=1, direction=COLUMN, padding=50, background_color="black"))
        self.main_window.content = main_box
        self.main_window.show()


def main():
    return HelloWorld()


The result images are like these:
![image](https://github.com/beeware/briefcase/assets/134466342/352e09c1-c111-4578-9daf-918bec44c28b)
![image](https://github.com/beeware/briefcase/assets/134466342/f964b045-3b43-4b77-8519-c5a6fa899a31)

**Reply by mhsmith** [reply]
It looks like we fixed this since the last stable release. To use the current development version, edit your pyproject.toml file and replace the `toga-android` line with this:
```
    "git+https://github.com/beeware/toga@d66b6bfa#subdirectory=core",
    "git+https://github.com/beeware/toga@d66b6bfa#subdirectory=android",
```

**Reply by ccinsz** [reply]
It is ok now. Thank you for your help.
